src/core/client/admin/components/PaginatedSelect/PaginatedSelect.tsx

```diff
@@ -66,42 +96,58 @@ const PaginatedSelect: FunctionComponent<Props> = ({
               )}
             </Dropdown>
           </IntersectionProvider>
-        </ClickOutside>
-      )}
-    >
-      {({ toggleVisibility, ref, visible }) => (
-        <Button
-          className={cn(styles.button, className)}
-          variant="flat"
-          adornmentLeft
-          color="mono"
-          onClick={toggleVisibility}
-          ref={ref}
-          uppercase={false}
-        >
-          {Icon && (
-            <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
-          )}
-          <Flex alignItems="center" className={styles.wrapper}>
-            {selected}
+        )}
+      >
+        {({ ref }) => (
+          <Flex
+            className={cn(styles.button, className)}
+            onClick={() => setIsPopoverVisible(true)}
+            ref={ref}
+            justifyContent="space-between"
+            aria-label={label}
+          >
+            {Icon && (
+              <ButtonSvgIcon className={styles.buttonIconLeft} Icon={Icon} />
+            )}
+            {isPopoverVisible && !!onFilter ? (
+              <Localized
+                id="admin-paginatedSelect-filter"
+                attrs={{ "aria-roledescription": true, "aria-label": true }}
+              >
+                <TextArea
+                  className={styles.filterInput}
+                  onChange={(e) => onFilter(e.target.value)}
+                  ref={filterRef}
+                  aria-label="Filter results"
+                />
+              </Localized>
+            ) : (
+              <Flex
+                alignItems="center"
+                className={styles.wrapper}
+                style={{
```

Comment:
✅

Reply:
I think the `className={styles.wrapper}` needs to be added back in here now

Reply:
I've replaced it, but just a heads up, the ellipses overflow doesn't work, and wasn't working before this PR because of of the way that the `<Flex>` component wraps its contents in an extra div. It's worth a fix but since it is a problem that has existed for along time I think it is outside of the scope of this ticket.